code/modules/medical/genetics/bioEffects/harmful.dm

```diff
@@ -588,6 +588,41 @@
 		if (prob(emote_prob))
 			L.emote(emote_type)
 
+/datum/bioEffect/colorblindness
```

Comment:
I haven't checked but I'm pretty sure as its written this whole thing might go away if you reconnect.

Reply:
In order to do it properly there should be something (check out mob_properties, I think priority based system would work the best unless someone wants to write color matrix compositions) that holds the intended client color and it is also set in mob/Login.

Reply:
Added a mob_property and a check on login to add the effect.

I didn't set it to be a priority property since there was only this effect to check for, and changing how the effect is applied through priorities doesn't seem within the scope of this PR. I'll probably end up doing that, though, if I hopefully get around to adding other color matrix effects. And maybe fixing the flockvision thing.

I also defined the matrix values in _setup. I tried defining the whole list there, and also plugging the define as it is now into something like mob.client.color = list(MATRIX_PROTANOPIA), but it just set client.color to "#0000FF" and made everything blue. I'm probably not understanding how lists work, but as it is now, it appears to work.

Reply:
looks decent enough now